kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/Coordinator.java

```diff
@@ -215,16 +222,31 @@ private void commitToTable(
       LOG.info("Nothing to commit to table {}, skipping", tableIdentifier);
     } else {
       if (deleteFiles.isEmpty()) {
-        AppendFiles appendOp = table.newAppend();
-        branch.ifPresent(appendOp::toBranch);
-        appendOp.set(snapshotOffsetsProp, offsetsJson);
-        appendOp.set(COMMIT_ID_SNAPSHOT_PROP, commitState.currentCommitId().toString());
-        if (vtts != null) {
-          appendOp.set(VTTS_SNAPSHOT_PROP, Long.toString(vtts));
-        }
-        dataFiles.forEach(appendOp::appendFile);
-        appendOp.commit();
+        Map<Integer, List<DataFile>> dataFilesBySpec = new HashMap<>();
+        dataFiles.forEach(dataFile -> {
+          List<DataFile> list = dataFilesBySpec.computeIfAbsent(dataFile.specId(), ignored -> Lists.newArrayList());
+          list.add(dataFile);
+        });
+        Integer maxSpecId = dataFilesBySpec.keySet().stream().max(Integer::compare).get();
+
+        Transaction transaction = table.newTransaction();
+        dataFilesBySpec.entrySet().stream().sorted(Comparator.comparingInt(Map.Entry::getKey)).forEach(entry -> {
+          AppendFiles appendOp = transaction.newAppend();
+          branch.ifPresent(appendOp::toBranch);
+          entry.getValue().forEach(appendOp::appendFile);
+          appendOp.set(COMMIT_ID_SNAPSHOT_PROP, commitState.currentCommitId().toString());
+          if (Objects.equals(entry.getKey(), maxSpecId)) {
+            appendOp.set(snapshotOffsetsProp, offsetsJson);
+            if (vtts != null) {
+              appendOp.set(VTTS_SNAPSHOT_PROP, Long.toString(vtts));
+            }
+          }
+          appendOp.commit();
+        });
+        transaction.commitTransaction();
       } else {
+        // TODO: can we use the same "commit data/delete files as separate snapshots, one snapshot for each PartitionSpec" trick here?
+        //  worried about edge cases.
```

Comment:
Open question: do we see any issues with using the same trick here? 
I'm less familiar with iceberg's "delete" APIs and am worried about whether there are edge-cases related to ordering of DeleteFiles that we need to be wary of.

Reply:
I _think_ we are ok from reading the spec.  Doing the same trick, I don't think we would violate the following:

```
A position delete file must be applied to a data file when all of the following are true:
-  The data file's data sequence number is less than or equal to the delete file's data sequence number
-  The data file's partition (both spec and partition values) is equal to the delete file's partition
An equality delete file must be applied to a data file when all of the following are true:
-  The data file's data sequence number is strictly less than the delete's data sequence number
-  The data file's partition (both spec id and partition values) is equal to the delete file's partition or the delete file's partition spec is unpartitioned
```

As long as you pair specId w/ id you should be fine.  Looking at how `IcebergWriterFactory` / `Utilities.java` construct `PartitionedDeltaWriter` you should be fine.  Inside of that class it constructs both a data file writer and an eq data file writer off the same `table.spec()`, so matching specIds should be fine.

Reply:
I agree, I've reached the same conclusion here independently, we should be safe, thanks for the support! :D

Reply:
I still think there is an ordering though.  Older files need to be committed first, regardless of partitionSpec.

This should be offset order out of the control topic?

Reply:
😓 Reverted the changes on the row-delta side for now, we can come back to this later.